src/main/java/com/internship/socialnetwork/controller/PostController.java

```diff
@@ -4,52 +4,54 @@
 import com.internship.socialnetwork.service.PostService;
 import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
 
 import static org.springframework.http.HttpStatus.CREATED;
 
 @RequiredArgsConstructor
-@RequestMapping("/api/v1/posts")
+@RequestMapping("/api/v1")
 @RestController
 public class PostController {
 
     private final PostService postService;
 
-    @GetMapping
+    @GetMapping("/users/{id}/posts")
```

Comment:
We can either have /users/.../posts or /posts in the same controller, we should not mix it. This one can be /posts with mandatory @RequestParam authorId then